lib/applications/field-resolvers.ts

```diff
@@ -43,11 +44,55 @@ export const applicationApplicantResolver: FieldResolver<
 
 /**
  * Fetch processing data of application
+ * Refresh temporary s3 application document url if it has expired.
  * @returns Application processing object
  */
 export const applicationProcessingResolver: FieldResolver<
   Application,
   Omit<ApplicationProcessing, 'invoice'>
 > = async (parent, _args, { prisma }) => {
-  return await prisma.application.findUnique({ where: { id: parent.id } }).applicationProcessing();
+  const application = await prisma.application
+    .findUnique({ where: { id: parent.id } })
+    .applicationProcessing();
+
+  if (!application) {
+    return null;
+  }
+
+  if (!process.env.APPLICATION_DOCUMENT_LINK_TTL_DAYS) {
+    throw new ApolloError('Application document link duration not defined');
+  }
+  // Update the signed S3 URL if it has expired.
+  // Get the valid duration period from env.
+  const applicationDocumentLinkDuration = parseInt(process.env.APPLICATION_DOCUMENT_LINK_TTL_DAYS);
+  const DAY = 24 * 60 * 60 * 1000;
+  const daysDifference =
+    Math.floor(new Date().getTime() / DAY) -
+    Math.floor(application?.documentsUrlUpdatedAt.getTime() / DAY);
+  if (daysDifference > applicationDocumentLinkDuration) {
+    let signedUrl;
+    try {
+      signedUrl = getSignedUrlForS3(application.documentsS3ObjectKey as string);
+    } catch (error) {
+      throw new ApolloError(`Failed to get application document URL: ${error}`);
+    }
+    try {
+      await prisma.application.update({
+        where: { id: application.id },
+        data: {
+          applicationProcessing: {
+            update: {
+              documentsUrl: signedUrl,
+            },
+          },
+        },
+      });
+    } catch {
+      throw new ApolloError('Failed to update temporary application document URL');
+    }
+    return await prisma.application
+      .findUnique({ where: { id: parent.id } })
+      .applicationProcessing();
+  }
```

Comment:
Since we're not saving the url in the db, then we don't need to update it on every read. Instead we will generate the signed url from the objectKey and add it to the response. You can use the code that you have in `lib/application-processing/resolvers.ts` to generate the signedUrl